plugins/collections/get.js

```diff
@@ -6,6 +6,42 @@ const schema = require('screwdriver-data-schema');
 const getSchema = schema.models.collection.get;
 const idSchema = joi.reach(schema.models.collection.base, 'id');
 
+/**
+ * Helper function to get last builds of pipelines
+ *
+ * @param   {Array}    pipelines     - array of pipeline models to get last builds for
+ * @param   {Factory}  eventFactory  - factory for getting last event model
+ * @returns {Array}
+ */
+function getPipelinesHealth(pipelines, eventFactory) {
+    return pipelines
+        .filter(pipeline => !!pipeline)
+        .map((pipeline) => {
+            const result = Object.assign({}, pipeline.toJson());
+
+            if (!pipeline.lastEventId) {
+                return result;
+            }
+
+            return eventFactory.get(pipeline.lastEventId)
+                .then((event) => {
+                    if (!event) {
+                        return result;
+                    }
+
+                    return event.getBuilds()
+                        .then((builds) => {
+                            if (builds.length) {
+                                result.lastBuilds = builds.map(b => b.toJson()).reverse();
+                            }
+
+                            return result;
+                        });
+                })
+                .catch(() => result);
```

Comment:
I'm not sure if it is very likely for it to fail but in the case that it does fail it could throw an error here. My thought process was that if an error is caused here by the `eventFactory` lookup, it should just return whatever data it has for the pipeline and not error out the whole thing and lead to the user receiving no data at all on the UI as this is the `GET` for the entire collection.